Library/Homebrew/cask/cask_loader.rb

```diff
@@ -186,6 +186,100 @@ def load(config:)
       end
     end
 
+    # Loads a cask from the JSON API.
+    class FromAPILoader
+      attr_reader :token, :path
+
+      FLIGHT_STANZAS = [:preflight, :postflight, :uninstall_preflight, :uninstall_postflight].freeze
+
+      def self.can_load?(ref)
+        Homebrew::API::Cask.all_casks.key? ref
+      end
+
+      def initialize(token)
+        @token = token
+        @path = CaskLoader.default_path(token)
+      end
+
+      def load(config:)
+        json_cask = Homebrew::API::Cask.all_casks[token]
+
+        if (bottle_tag = ::Utils::Bottles.tag.to_s.presence) &&
+           (variations = json_cask["variations"].presence) &&
+           (variation = variations[bottle_tag].presence)
+          json_cask = json_cask.merge(variation)
+        end
+
+        json_cask.deep_symbolize_keys!
+
+        # Use the cask-source API if there are any `*flight` blocks
+        if json_cask[:artifacts].any? { |artifact| FLIGHT_STANZAS.include?(artifact.keys.first) }
```

Comment:
Not for this PR: I wonder if it might be nice to push this to the API instead/as well so we have something like `needs_cask_source: true` in the JSON?

Reply:
Do you think we want to remove the `*flight` block source that was added in #14324 from the API if we're not actually going to use it?